conn.go

```diff
@@ -791,6 +795,16 @@ func marshalQueryValue(typ TypeInfo, value interface{}, dst *queryValues) error
 }
 
 func (c *Conn) executeQuery(qry *Query) *Iter {
+	ctx := qry.context
+	if rt, ok := qry.rt.(RetryPolicyWithAttemptTimeout); ok && rt.AttemptTimeout() > 0 {
+		if ctx == nil {
+			ctx = context.Background()
```

Comment:
Tbh we should just move this all the way up to default on Query, when we do an API change I want to change all the frontend functions to take context as their first param. Ill make an issue for this, dont worry about changing it now

Reply:
on query/batch